ufl/core/external_operator.py

```diff
@@ -0,0 +1,98 @@
+from ufl.utils.str import as_native_str
+from ufl.coefficient import Coefficient
+from ufl.core.ufl_type import ufl_type
+from ufl.constantvalue import as_ufl
+from ufl.log import error
+from ufl.finiteelement.mixedelement import TensorElement
+from ufl.functionspace import FunctionSpace
+from ufl.domain import default_domain
+import weakref
+
+
+@ufl_type(inherit_indices_from_operand=0, is_terminal=True, is_differential=True)
+class ExternalOperator(Coefficient):
+
+    # Slots are disabled here because they cause trouble in PyDOLFIN
+    # multiple inheritance pattern:
+    # __slots__ = as_native_strings(("ufl_operands", "_ufl_function_space", "derivatives", "_ufl_shape"))
+    _ufl_noslots_ = True
+    _ufl_all_external_operators_ = weakref.WeakKeyDictionary()
+
+    def __init__(self, *operands, function_space, derivatives=None, count=None):
+
+        # Check operands
+        for i in operands:
+            as_ufl(i)
+
+        self.ufl_operands = operands
```

Comment:
Prefer `tuple(map(as_ufl, operands))`. Otherwise you might even end up with non-UFL objects in the operands (e.g. if one of them was an integer).

Reply:
Ok thank you, I will do that !